presto-product-tests/src/main/java/io/prestosql/tests/hive/acid/TestPartitionedInsertOnlyAcidTable.java

```diff
@@ -0,0 +1,214 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.tests.hive.acid;
+
+import com.google.common.net.HostAndPort;
+import com.google.inject.Inject;
+import io.prestosql.plugin.hive.authentication.NoHiveMetastoreAuthentication;
+import io.prestosql.plugin.hive.metastore.thrift.ThriftHiveMetastoreClient;
+import io.prestosql.plugin.hive.metastore.thrift.Transport;
+import io.prestosql.tempto.ProductTest;
+import io.prestosql.tempto.Requirement;
+import io.prestosql.tempto.RequirementsProvider;
+import io.prestosql.tempto.configuration.Configuration;
+import io.prestosql.tempto.context.ThreadLocalTestContextHolder;
+import io.prestosql.tempto.fulfillment.table.MutableTablesState;
+import io.prestosql.tempto.fulfillment.table.hive.HiveTableDefinition;
+import io.prestosql.tempto.query.QueryExecutor;
+import io.prestosql.tempto.query.QueryResult;
+import org.apache.hadoop.hive.metastore.api.AllocateTableWriteIdsRequest;
+import org.apache.hadoop.hive.metastore.api.Table;
+import org.apache.thrift.TException;
+import org.testng.annotations.Test;
+
+import java.net.URI;
+import java.util.Collections;
+import java.util.Optional;
+import java.util.UUID;
+
+import static io.prestosql.tempto.Requirements.compose;
+import static io.prestosql.tempto.assertions.QueryAssert.Row.row;
+import static io.prestosql.tempto.assertions.QueryAssert.assertThat;
+import static io.prestosql.tempto.fulfillment.table.TableRequirements.mutableTable;
+import static io.prestosql.tempto.query.QueryExecutor.query;
+import static io.prestosql.tests.TestGroups.HIVE_ACID;
+import static io.prestosql.tests.TestGroups.PROFILE_SPECIFIC_TESTS;
+
+public class TestPartitionedInsertOnlyAcidTable
```

Comment:
TestHiveTransactionalTable has complicated things. Insert-only transactional tables will work with hdp3.1 images but fail with hdp2.6 image.

These tests supersede  TestHiveTransactionalTable tests and cannot be moved to existing tests groups as explained earlier. Should I just delete TestHiveTransactionalTable?

Reply:
I have changed TestHiveTransactionalTable to TestHiveTransactionalTableInsert and it now only tests writes to transactional tables. This is done to ensure default tests continue to work.

TestInsertOnlyAcidTableRead is now added, parametrized on partitioning.

TestFullAcidTableRead is separately added for full acid reads, this will be extended when full acid changes land.

Reply:
it's possible to check hive major version in the test and i think it results in cleaner test code management than group-based only.
(i am not sure if i've already merged this to master or it's on hdp3 branch yet)

Reply:
I can use `onHive().getConnection().getMetaData().getDatabaseMajorVersion()` and run these tests if version >= 3

Reply:
Removed the group